store/store.go

```diff
@@ -231,6 +231,10 @@ func (s *Store[H]) GetByHeight(ctx context.Context, height uint64) (H, error) {
 		return zero, errors.New("header/store: height must be bigger than zero")
 	}
 
+	if h, err := s.getByHeight(ctx, height); err == nil || ctx.Err() != nil {
```

Comment:
Do we care about ctx error here?

Reply:
well yeah, if ctx is cancelled no need to subscribe (will fail anyway)

Reply:
Yeah, that's what I mean, so this check is redundant imo

Reply:
Done